src/components/Search/SearchLayout.tsx

```diff
@@ -119,6 +137,17 @@ export function SearchLayout({
   const [sidebarOpen, setSidebarOpen] = useState(false);
   const ctx = useMemo(() => ({ sidebarOpen, setSidebarOpen }), [sidebarOpen]);
 
+  const router = useRouter();
+  const { trackSearch } = useTrackEvent();
+
+  useEffect(() => {
+    const result = searchQuerySchema.safeParse(router.query);
+    if (!result.success) return;
+
+    const { query, sortBy: index, ...filters } = result.data;
+    if (query && index) trackSearch({ query, index, filters });
```

Comment:
tbf: Note that this might get triggered on pretty much partial searches since the user will be typing and it may trigger a partial search.

Reply:
I think it's okay in this case since people tend to do partial queries anyways when looking for something. Unless we enforce hitting Enter to start a new search term